ls1intum/Artemis
# [Feature] Improve automated grading page (former manage test-cases)

### Checklist
- [x] Client: I updated multiple integration tests (Jest) related to the features

### Motivation and Context



Prepare grading page to incorporate static code analysis configuration.

### Description


1. Rename test-case components to configure-grading components
2. Use inline-editing for editable fields
2. Add tabbed routing for TestCases and CodeAnalysis Subpages
4. Load example category mapping in grading page

The static code analysis configuration is not yet linked to the backend, so any changes there have no effect. This PR functions merely as a basis for the planned implementation.

### Steps for Testing


1. Log in to Artemis
2. Create a new Programming Exercise with Static Code Analysis enabled.
3. Navigate to Course Administration -> Programming Exercise -> Configure Grading
4. Check that the test-case configuration still works (change weights, saving, resetting, re-evaluating results, etc.)
5. Check that the static code analysis categories are displayed correctly.

### Test Coverage

- **programming-exercise-configure-grading.component.ts**: 83%
- **programming-exercise-configure-grading-actions.component.ts**: 100%
- **table-editable-field.component.ts**: 100%

### Screenshots


<img width="294" alt="Bildschirmfoto 2020-08-18 um 09 25 52" src="https://user-images.githubusercontent.com/13920539/90483311-d4070400-e134-11ea-9d4b-4395ed8864c4.png">
<img width="1431" alt="Bildschirmfoto 2020-09-01 um 10 31 24" src="https://user-images.githubusercontent.com/13920539/91827724-8368ce00-ec3f-11ea-86a0-23f6e5eac7f7.png">
<img width="1422" alt="Bildschirmfoto 2020-09-01 um 13 17 51" src="https://user-images.githubusercontent.com/13920539/91843163-94243e80-ec55-11ea-8632-59e0e367999e.png">
